core/trino-main/src/main/java/io/trino/connector/system/jdbc/ColumnJdbcTable.java

```diff
@@ -258,9 +259,9 @@ public RecordCursor cursor(ConnectorTransactionHandle transactionHandle, Connect
                 continue;
             }
 
-            if ((schemaDomain.isAll() && tableDomain.isAll()) || (schemaFilter.isPresent() && tableFilter.isPresent())) {
-                QualifiedTablePrefix tablePrefix = tablePrefix(catalog, schemaFilter, tableFilter);
-                Map<SchemaTableName, List<ColumnMetadata>> tableColumns = listTableColumns(session, metadata, accessControl, tablePrefix);
+            Optional<QualifiedTablePrefix> simpleTablePrefix = tablePrefixIfPossible(catalog, schemaFilter, tableFilter);
+            if ((schemaDomain.isAll() && tableDomain.isAll()) || simpleTablePrefix.isPresent()) {
```

Comment:
if `schemaDomain` and `tableDomain` are non-singlevalue and non-all, this wouldn't filter at all right? do we plan to rely on FilterNode as the constraints aren't necessarily "enforced"?

Reply:
We are already relying on `FilterNode` here. https://github.com/trinodb/trino/blob/master/core/trino-main/src/main/java/io/trino/connector/system/SystemTablesMetadata.java#L178 is returning `constraint.getSummary()` as `remainingFilter`

Reply:
yes, meant to point out that seems a bit different from current behavior of the method. A comment would help indicate that, but would leave it upto you.